spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicConfigurationSupport.java

```diff
@@ -0,0 +1,369 @@
+		DestinationTopicResolver destinationTopicResolver = this.componentFactory.destinationTopicResolver();
+		if (destinationTopicResolver instanceof DefaultDestinationTopicResolver) {
+			DefaultDestinationTopicResolver ddtr = (DefaultDestinationTopicResolver) destinationTopicResolver;
+			NonBlockingRetriesConfigurer configurer = new NonBlockingRetriesConfigurer();
+			configureNonBlockingRetries(configurer);
+			if (configurer.clearDefaultFatalExceptions) {
+				ddtr.setClassifications(new HashMap<>(), true);
+			}
+			if (configurer.addToFatalExceptions != null) {
+				Stream.of(configurer.addToFatalExceptions).forEach(ddtr::addNotRetryableExceptions);
+			}
+			if (configurer.removeFromFatalExceptions != null) {
+				Stream.of(configurer.removeFromFatalExceptions).forEach(ddtr::removeClassification);
+			}
+		}
+		Consumer<DestinationTopicResolver> resolverConsumer = customizeDestinationTopicResolver();
+		Assert.notNull(resolverConsumer, "customizeDestinationTopicResolver must not return null");
+		return destinationTopicResolver;
+	}
+
+	/**
+	 * Override this method to configure the {@link DestinationTopicResolver}.
+	 * @return a {@link DestinationTopicResolver} consumer.
+	 */
+	protected Consumer<DestinationTopicResolver> customizeDestinationTopicResolver() {
+		return dtr -> {
+		};
+	}
+
+	/**
+	 * Provides the {@link KafkaConsumerBackoffManager} instance.
+	 * Override this method to provide a customized implementation.
+	 * A {@link PartitionPausingBackOffManagerFactory} can be used for that purpose,
+	 * or a different implementation can be provided.
+	 * @param registry the {@link ListenerContainerRegistry} to be used to fetch the
+	 * {@link MessageListenerContainer} to be backed off.
+	 * @return the instance.
+	 */
+	@Bean(name = KafkaListenerConfigUtils.KAFKA_CONSUMER_BACK_OFF_MANAGER_BEAN_NAME)
+	public KafkaConsumerBackoffManager kafkaConsumerBackoffManager(
+			@Qualifier(KafkaListenerConfigUtils.KAFKA_LISTENER_ENDPOINT_REGISTRY_BEAN_NAME)
+					ListenerContainerRegistry registry) {
+		PartitionPausingBackOffManagerFactory factory = new PartitionPausingBackOffManagerFactory(registry);
+		this.disposableBackOffManagerFactory = factory;
+		return factory.create();
+	}
+
+	/**
+	 * Override this method if you want to provide a subclass
+	 * of {@link RetryTopicComponentFactory} with different
+	 * component implementations or subclasses.
+	 * @return the instance.
+	 */
+	protected RetryTopicComponentFactory createComponentFactory() {
+		return new RetryTopicComponentFactory();
+	}
+
+	@Deprecated
+	@Bean(name = RetryTopicInternalBeanNames.RETRY_TOPIC_BOOTSTRAPPER)
+	RetryTopicBootstrapper retryTopicBootstrapper(ApplicationContext context) {
+		return new RetryTopicBootstrapper(context, context.getAutowireCapableBeanFactory());
+	}
+
+	@Override
+	public void destroy() throws Exception {
+		if (this.disposableBackOffManagerFactory != null) {
+			this.disposableBackOffManagerFactory.destroy();
```

Comment:
Therefore when we revise an `executor` for the `WakingKafkaConsumerTimingAdjuster`, we won't need this logic over here at all.